core/node/libp2p/rcmgr.go

```diff
@@ -0,0 +1,368 @@
+			limitFile, err := os.Open(limitFilePath)
+			switch {
+			case err == nil:
+				defer limitFile.Close()
+				limiter, err = rcmgr.NewLimiterFromJSON(limitFile, defaultLimits)
+				if err != nil {
+					return nil, opts, fmt.Errorf("error parsing libp2p limit file: %w", err)
+				}
+			case errors.Is(err, os.ErrNotExist):
+				limiter = rcmgr.NewStaticLimiter(defaultLimits)
+			default:
+				return nil, opts, err
+			}
+
+			setDefaultServiceLimits(limiter) // see rcmgr_defaults.go
+
+			ropts := []rcmgr.Option{rcmgr.WithMetrics(createRcmgrMetrics())}
+
+			if os.Getenv("LIBP2P_DEBUG_RCMGR") != "" {
+				traceFilePath := filepath.Join(repoPath, NetLimitTraceFilename)
+				ropts = append(ropts, rcmgr.WithTrace(traceFilePath))
+			}
+
+			manager, err = rcmgr.NewResourceManager(limiter, ropts...)
+			if err != nil {
+				return nil, opts, fmt.Errorf("error creating libp2p resource manager: %w", err)
+			}
+
+		} else {
+			log.Debug("libp2p resource manager is disabled")
+			manager = network.NullResourceManager
+		}
+
+		opts.Opts = append(opts.Opts, libp2p.ResourceManager(manager))
+
+		lc.Append(fx.Hook{
+			OnStop: func(_ context.Context) error {
+				return manager.Close()
+			}})
+
+		return manager, opts, nil
+	}
+}
+
+type NetStatOut struct {
+	System    *network.ScopeStat           `json:",omitempty"`
+	Transient *network.ScopeStat           `json:",omitempty"`
+	Services  map[string]network.ScopeStat `json:",omitempty"`
+	Protocols map[string]network.ScopeStat `json:",omitempty"`
+	Peers     map[string]network.ScopeStat `json:",omitempty"`
+}
+
+func NetStat(mgr network.ResourceManager, scope string) (NetStatOut, error) {
+	var err error
+	var result NetStatOut
+	switch {
+	case scope == "all":
+		rapi, ok := mgr.(rcmgr.ResourceManagerState)
+		if !ok { // NullResourceManager
+			return result, NoResourceMgrError
+		}
+
+		stat := rapi.Stat()
+		result.System = &stat.System
+		result.Transient = &stat.Transient
+		if len(stat.Services) > 0 {
+			result.Services = stat.Services
+		}
+		if len(stat.Protocols) > 0 {
+			result.Protocols = make(map[string]network.ScopeStat, len(stat.Protocols))
+			for proto, stat := range stat.Protocols {
+				result.Protocols[string(proto)] = stat
+			}
+		}
+		if len(stat.Peers) > 0 {
+			result.Peers = make(map[string]network.ScopeStat, len(stat.Peers))
+			for p, stat := range stat.Peers {
+				result.Peers[p.Pretty()] = stat
+			}
+		}
+
+		return result, nil
+
+	case scope == config.ResourceMgrSystemScope:
+		err = mgr.ViewSystem(func(s network.ResourceScope) error {
+			stat := s.Stat()
+			result.System = &stat
+			return nil
+		})
+		return result, err
+
+	case scope == config.ResourceMgrTransientScope:
+		err = mgr.ViewTransient(func(s network.ResourceScope) error {
+			stat := s.Stat()
+			result.Transient = &stat
+			return nil
+		})
+		return result, err
+
+	case strings.HasPrefix(scope, config.ResourceMgrServiceScopePrefix):
+		svc := scope[4:]
```

Comment:
Can we use `svc := strings.TrimPrefix(scope, config.ResourceMgrServiceScopePrefix)` for this and the other scope prefixes?